### ISSUE TITLE
wrong timestamps of transaction

### ISSUE THREAD

**Issue by beulemann** [PaycoinFoundation/paycoin#139]
the time shown in the transactions tab, is the time, when the transaction was recieved by the wallet.

could it be possible to use the "real" transaction time?
(the time, when the transaction is added to a block)

**Comment by mitchellcash** [comment]
Yeah I think this is doable. There will be some exceptions but I'll look into it for you. 

I can't remember does rescan get the block times?

**Comment by IngCr3at1on** [comment]
@mitchellcash each transaction has a time stored with it itself usually referred to in the code as nTime, should be able to get it directly from the transaction.

**Comment by IngCr3at1on** [comment]
Well... I'm not entirely sure this is possible (I may look into it more but this was from my first attempt)....

![selection_071](https://cloud.githubusercontent.com/assets/922158/9279896/34e104e6-4280-11e5-97c3-b5420806e66e.png)

**Comment by beulemann** [comment]
why are there problems implementing it?
the dogecoin wallet shows the "correct" timestamps.

**Comment by mitchellcash** [comment]
I think there is a solution. I just haven't had the chance to investigate further right now.

**Comment by esipas** [comment]
this would be a neat fix :)

**Comment by IngCr3at1on** [comment]
Basically the reason this can't be implemented at this time is because the times that @beulemann is wanting displayed aren't currently stored in the walletdb.

These requirements were not implemented until bitcoin 7.0 which is probably why you see them in some coins but not in others....

See https://github.com/bitcoin/bitcoin/pull/1393 for more information.
This is also required to properly support salvaging corrupt wallets, in other words I'm already working on it as I write this (there's a chance I'll stumble across something else required to implement it, meaning I have no idea how long it will take).

After implementing the above changing the QT display should be rather easy.
